lib/events.js

```diff
@@ -54,64 +89,104 @@ $(document).ready(function() {
-                        meal.appendToTable($bfastIndex)
-                        meal.mealCalories($brfastCal)
-                        meal.mealCaloriesRemaining($brfastRemainingCal, bfastTarget)
+                        meal.populateTable($bfastIndex, $brfastCal, $brfastGoalCal, $brfastRemainingCal, bfastTarget)
                         break;
                     case "Lunch":
-                        meal.appendToTable($lunchIndex)
-                        meal.mealCalories($lunchCal)
-                        meal.mealCaloriesRemaining($lunchRemainingCal, bfastTarget)
+                        meal.populateTable($lunchIndex, $lunchCal, $lunchGoalCal, $lunchRemainingCal, lunchTarget)
                         break;
                     case "Snack":
-                        meal.appendToTable($snackIndex)
-                        meal.mealCalories($snackCal)
-                        meal.mealCaloriesRemaining($snackRemainingCal, bfastTarget)
+                        meal.populateTable($snackIndex, $snackCal, $snackGoalCal, $snackRemainingCal, snackTarget)
                         break;
                     case "Dinner":
-                        meal.appendToTable($dinnerIndex)
-                        meal.mealCalories($dinnerCal)
-                        meal.mealCaloriesRemaining($dinnerRemainingCal, bfastTarget)
+                        meal.populateTable($dinnerIndex, $dinnerCal, $dinnerGoalCal, $dinnerRemainingCal, dinnerTarget)
                         break;
                 }
             })
             return meals
         })
-        .then(function(meals) {
-            let total = meals.reduce(function(sum, meal) {
-                return sum + meal.total
-            }, 0)
-            let remaining = target - total
-            $totalCal.text(total)
-            $remainingCal.text(remaining)
-            $goalCal.text(target)
-            if (remaining < 0) {
-                $remainingCal.addClass('negative')
-            } else {
-                $remainingCal.removeClass('negative')
-            }
+        .then(function() {
+            Meal.updateGrandTotal(target)
         })
 
     $foodSubmit.on('click', function() {
         let food = {
             name: $newFoodName.val(),
             calories: $newFoodCalories.val()
         }
-        if(Food.validate(food)) {
-          Food.addNewFood(food, $foodIndex);
-          $newFoodName.val('');
-          $newFoodCalories.val('');
-          $('.notify').remove();
+
+        if (Food.validate(food)) {
+            Food.addNew(food, $foodIndex);
+            $newFoodName.val('');
+            $newFoodCalories.val('');
+            $('.notify').remove();
         }
     })
 
     $foodIndex.on('click', '.delete', function(event) {
         var food = this.parentElement.parentElement
         var id = food.id
-        Food.deleteFood(id)
+        Food.delete(id)
             .then(function() {
                 $(`#${id}`).fadeOut(300);
```

Comment:
fancy fadeOut! are absolutely sure 300 milliseconds is the right length of time? :)